core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -412,6 +412,8 @@ private static class MetricsRequestEventListener implements RequestEventListener
     public MetricsRequestEventListener(final Map<Method, RequestScopedMetrics> metrics, Time time) {
       this.metrics = metrics;
       this.time = time;
+      // CIAM-2673: if an exception occurs, MATCHING_START is never reached, resulting in false latency metrics
+      this.started = time.milliseconds();
```

Comment:
High level I think this should work, but I would like to confirm when the EventsListener is instantiated. Please add tests to verify this works correctly.

https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/ApplicationEventListener.html

It looks like it should be created once per request at RequestEvent.Type.START

Reply:
Is this better than doing it after line 420?

Reply:
RequestEvent.Type.START won't be passed there: https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/RequestEventListener.html

`RequestEvent.Type.Start` is only handled by the ApplicationEventListener in the `onRequest` method.